packages/web/src/common/hooks/useFeatureFlag.ts

```diff
@@ -1,44 +1,78 @@
-import { useMemo } from 'react'
-
-import { useSelector } from 'react-redux'
+import { useCallback, useEffect, useMemo, useState } from 'react'
 
 import {
   FeatureFlags,
-  FeatureFlagCohortType,
-  RemoteConfigInstance,
-  flagCohortType
+  RemoteConfigInstance
 } from 'common/services/remote-config'
-import { getAccountUser } from 'common/store/account/selectors'
-import { isRemoteConfigLoaded } from 'common/store/remote-config/selectors'
-import { StateWithRemoteConfig } from 'common/store/remote-config/slice'
 
 export const FEATURE_FLAG_OVERRIDE_KEY = 'FeatureFlagOverride'
 
 export type OverrideSetting = 'enabled' | 'disabled' | null
 
+/**
+ * Helper for when to recompute flag state, used by both FeatureFlags
+ * and RemoteConfig. Recomputes when:
+ * - User logs in (account is seen in store)
+ * - Config loads
+ * - User ID is set on Optimizely (seen by event emission)
+ **/
+const useRecomputeBool = (
```

Comment:
I think this overly abstracts the recomputation to the point that it's harder to grok - `shouldRecompute` to me indicates it recomputes when it's _true_, not that it's being toggled to serve as a proxy for these things changing. Wouldn't it make more sense to put `hasAccount`, `configLoaded` and something for the state change/user id listener in the `useMemo` dependency list directly?

I'm also not seeing it shared in `useRemoteVar` despite the comment seeming to indicate it?

Reply:
Totally forgot to include this in useRemoteVar! I'll change the name to make this clearer - I didn't want to have to implement the same stuff for the user state change in both places, feels a bit error prone